workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkerTester.kt

```diff
@@ -1,25 +1,33 @@
-@file:OptIn(FlowPreview::class, ExperimentalCoroutinesApi::class)
+@file:OptIn(ExperimentalCoroutinesApi::class)
 
 package com.squareup.workflow1.testing
 
+import app.cash.turbine.Event.Item
+import app.cash.turbine.test
 import com.squareup.workflow1.Worker
 import com.squareup.workflow1.testing.WorkflowTestRuntime.Companion.DEFAULT_TIMEOUT_MS
-import kotlinx.coroutines.Dispatchers.Unconfined
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.cancelChildren
-import kotlinx.coroutines.channels.ReceiveChannel
-import kotlinx.coroutines.flow.produceIn
-import kotlinx.coroutines.plus
-import kotlinx.coroutines.runBlocking
-import kotlinx.coroutines.supervisorScope
-import kotlinx.coroutines.withTimeout
-import kotlinx.coroutines.yield
+import kotlinx.coroutines.test.TestCoroutineScheduler
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.withTimeoutOrNull
+import kotlin.time.DurationUnit.MILLISECONDS
+import kotlin.time.toDuration
 
 public interface WorkerTester<T> {
 
+  /**
+   * Access the [TestCoroutineScheduler] of the [kotlinx.coroutines.test.TestScope] running
+   * the [Worker]'s [test].
+   *
+   * This can be used to advance virtual time for the [CoroutineDispatcher] that the the Worker's
+   * flow is flowing on.
+   */
+  public val testCoroutineScheduler: TestCoroutineScheduler
```

Comment:
This is added to control virtual time on the scope in which the Turbine is launched and thus where the Worker's upstream flow is producing values.